### DISCUSSION TITLE
Is there a new package going to be released?

### DISCUSSION THREAD

**Discussion by zhimin-z** [dsgibbons/shap#11] [Q&A] [answered]
Refer to https://github.com/slundberg/shap/pull/1890 for running error

**Comment by thatlittleboy** [comment, marked as answer]
**EDIT (2023-06-02)**: slundberg has replied, please see slundberg#2942 discussion for follow-ups

---

Hi @zhimin-z , thank you for raising the discussion point (converted to a discussion because it's not an issue). 

Releasing a new package (under a new name): That's definitely one of the options available to us, I think. You may read the general discussion on the purpose of this repo here: https://github.com/dsgibbons/shap/discussions/1 if you haven't. It's covered somewhat in the main post.

My vision (which may or may not be fully aligned with what @dsgibbons has in mind) is:
* fix all the tests & deprecation warnings (#4)
* introduce linting & enforcing style standards via black/pre-commit et al.
* test and support for python 3.11 (issue slundberg#2904)
* docs are available and rendering correctly
* add and build a changelog prior to the release (#47)

These are the minimal conditions under which I'll start considering next steps, which will either be 1) merging this back into the main `shap` repo (with slundberg's blessing to invite more collaborators to ensure there can be continued active support on that repo), or 2) release a new package under a new name if (1) fails. @dsgibbons  feel free to chime in here if you disagree / have more thoughts on this.

Option 2 is obviously less preferable since we would need to spend time doing PR / redirecting users to the new package.

Some more nice-to-haves that I'm thinking about:
* add a CONTRIBUTING guideline to help new contributors. Any new features and bug fixes should be accompanied by corresponding test(s).
* add a Github issue template, as we expect more questions and issues to stream in. E.g., for issues, we should encourage the bug reporter to provide a reproducible example.

We would aim to port over some of the PRs (especially some of the simpler ones) in the main repo once tests have passed here. Maybe those can sneak into the new release too.

---

As to your running error, as already mentioned by other posters, the fix has already been merged in (#7 ).

**Reply by connortann** [reply]
I'm a big fan of releasing early & often, rather than delaying until every issue is fixed. We have already fixed a number of major bugbears, and it seems folk are already keen to get their hands on the package.

Releasing early and often would also enable us to implement & test out a more automated release pipeline.

If we decide to publish on PyPI, shall we aim to make early Beta or [Release candidate](https://devguide.python.org/developer-workflow/development-cycle/#release-candidate-rc) releases?

**Reply by thatlittleboy** [reply]
I think we can try a beta release soon / now. As soon as we come to a conclusion on option 1 or 2.

In general, I'm also agreeable on "release early & often", up to a certain extent though. (e.g. we wouldn't want to release after every commit, just to take things to the extreme. xD)

**Comment by stephenpardy** [comment]
As an active user of SHAP I want to thank you for pushing this ahead. I am really excited for what everyone here is working on and the effort to keep this great package alive. I would really love to see a new package deployed because for some of the tools I work on I rely on having packages available through pypi as opposed to github. I imagine that I am not alone in this need. 

I am willing to help and contribute as needed. Thanks again!

**Reply by thatlittleboy** [reply]
Thanks! Feel free to raise issues on what bugs/issues you are most concerned about and want to see fixed; or if you're feeling adventurous, then take a stab at a PR if you'ld like. 😄
